dashboard/app/asset_storage.go

```diff
@@ -0,0 +1,347 @@
+// Copyright 2022 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/google/syzkaller/dashboard/dashapi"
+	"github.com/google/syzkaller/pkg/asset"
+	"golang.org/x/net/context"
+	"google.golang.org/appengine/v2"
+	db "google.golang.org/appengine/v2/datastore"
+	"google.golang.org/appengine/v2/log"
+)
+
+// TODO: decide if we really want to save job-related assets right now.
+//
+// The problem is that:
+// 1. We cannot first report the job status and then upload the assets -- an email will already
+//    be sent by that time.
+// 2. If we first upload assets and then construct a JobDoneReq response already with assets,
+//    we need to handle extra corner cases:
+//    - If JobDoneReq failed, assets must be deleted.
+//    - If we uploaded several files, all must be deleted.
+//    - What if syz-ci was killed somewhere in between? How do we get rid of those abadoned files?
```

Comment:
Are all of the cases automatically handled by the current algorithm?
We will see these abandoned assets are not needed anymore and automatically delete them after some time with no additional code, no?